packages/convenient_test_dev/lib/src/support/compile_time_config.dart

```diff
@@ -1,8 +1,21 @@
+import 'package:convenient_test_common/convenient_test_common.dart';
+
 class CompileTimeConfig {
-  static final kAppCodeDir = () {
-    const key = 'CONVENIENT_TEST_APP_CODE_DIR';
-    const value = String.fromEnvironment(key);
-    if (value.isEmpty) throw Exception('Please provide `$key` via --dart-define');
-    return value;
-  }();
+  static String _getConfigFromEnv(String key, String defaultVal, {bool required = false}) {
+    final value = String.fromEnvironment(key);
+
+    if (value.isEmpty && required) {
+      throw Exception('Please provide `$key` via --dart-define');
+    }
+
+    return value.isEmpty ? defaultVal : value;
+  }
+
+  static String get kAppCodeDir => _getConfigFromEnv('CONVENIENT_TEST_APP_CODE_DIR', '', required: true);
+
+  static String get dConvenientTestManagerHost =>
+      _getConfigFromEnv('CONVENIENT_TEST_MANAGER_HOST', kConvenientTestManagerHost);
```

Comment:
nit: Now we have CompileTimeconfig.kConvenientTestManagerHost and kConvenientTestManagerHost, which is confusing to readers. Maybe we should rename the old kConvenientTestManagerHost into something like *default*ConvenientTestManagerHost or *fallback*

Reply:
Got it, I can make that change. One thing I am concerned with is that `consts.dart` is in `convenient_test_common `so it is imported to the client and manager. 
However `compile_time_config.dart` is only in `convenient_test_dev` and so it is only imported into the client.

This means the "--dart-define" version is only available to the client and not both. This is a problem for `packages\convenient_test_manager_dart\lib\services\convenient_test_manager_service.dart` L23. (The server choosing a port).

Do you mind if I consolidate `compile_time_config.dart` into `consts.dart` so that the "--dart-define" logic is available to both client and manager.

Reply:
> Do you mind if I consolidate compile_time_config.dart into consts.dart so that the "--dart-define" logic is available to both client and manager.

LGTM as long as it does not have logic specific to client